### ISSUE TITLE
Missing battlefield obstacles

### ISSUE THREAD

**Issue by misiokles** [vcmi/vcmi#958]
In h3sprite.lod there few obstacles not configured and indexed in obstacles.json
These 2 obstacles (ObBhS04.def and ObBhS16a.def)

![OBBHL16](https://user-images.githubusercontent.com/29593042/191452109-d7b51d91-7757-4191-914b-618080f144a5.jpg)
![ObBhS16a](https://user-images.githubusercontent.com/29593042/191452196-93287da3-7d17-4c34-8e4b-7a48fe271f9e.jpg)
 are parts of special obstacle OBBHL16.pcx
 
![OBBHL16](https://user-images.githubusercontent.com/29593042/191452372-9e4f9470-7d9e-4e1a-8345-70f80226f565.jpg)
but those two shouldn't be spawned as separate small obstacles also?

Other not configured obstacles:

ObDSM02.def
![ObDSM02](https://user-images.githubusercontent.com/29593042/191452976-57b1c469-669d-4012-bd90-60ed195a0f90.jpg)

ObGRk02.def
![ObGRk02](https://user-images.githubusercontent.com/29593042/191453154-ae95118e-5cfa-4873-bbf3-f16b79d92f8a.jpg)

ObGrS03.def
![ObGrS03](https://user-images.githubusercontent.com/29593042/191453273-c604a0f8-8694-44bd-ba2b-ee756f6619d2.jpg)

ObRgS06.def
![ObRgS06](https://user-images.githubusercontent.com/29593042/191453362-746cb329-9295-4d6b-95c8-0c9a60c8178d.jpg)

Of course, these obstacles can be add to the game as separate mod, because they didn't included to the json for some reasons.

**Comment by misiokles** [comment]
And in obstacles.json in "specialBattlefields" id: 18 seems to be duplicate id: 21

**Comment by misiokles** [comment]
[missingObstacles.zip](https://github.com/vcmi/vcmi/files/9648577/missingObstacles.zip)
Here's my small mod with tose obstacles but poorly tested (it means I didn't observe their appearing)

**Comment by Opuszek** [comment]
TLDR: those seem to be non-issues except for ObGrS03.def, that I cannot really tell. 

1.  I tested over twenty beach combat maps in OG and neither ObBhS04.def nor ObBhS16a.def were ever used as obstacles by themselves (when I got three repetitions of OBBHL16.pcx and 5+ of each common small obstacle).

2. ObDSM02.def and ObGRk02.def are used by VCMI, just by different aliases.
https://github.com/user-attachments/assets/77c4c549-d176-4cbf-aea3-de27ff5854ba

https://github.com/user-attachments/assets/466fa9e9-5c22-4c8a-818c-d3499b0b17b2

My guess is that OG had to have slightly moved version because of some engine limitation that VCMI does not have.

3.  With tens of repetitions I could not find ObRgS06.def on neither rough nor cursed ground in OG when all other ObRgSx.def objects were shown multiple times. It is either a dead asset or requires some special conditions to be used.
4. ObGrS03.def - I have no idea. All those cliffs look similar, especially when green borders of the graphic blend with the green field. I am getting insane from trying to compare them.

**Comment by IvanSavenko** [comment]
Closing - WAD
